Adds the bundled feature. (#581)
parallaxsecond/rust-tss-esapi: tss-esapi-sys/README.md

generation of bindings. All the committed bindings **MUST** be generated from
the library version found under the `vendor` submodule.

## Bundling TPM-TSS

> tpm-tss is used by this library to communicate with TPMs. If this library

Reviewer: ```suggestion
[`tpm-tss`](https://github.com/tpm2-software/tpm2-tss) is used by this library to communicate with TPMs. If this library
```

is not available on your system you may optionally bundle (vendor) tpm-tss
during builds. tpm-tss can be provided from a local source path with the
environment variable `TPM_TSS_SOURCE_PATH` or it will be retrieved from

> github during the build. The version to retrieve can be controlled by setting

Reviewer: ```suggestion
Github during the build. The version to retrieve can be controlled by setting
```

the `TPM2_TSS_SOURCE_VERSION` environment variable.

> N.B. On windows it might be necessary to manually create the VERSION file
> when a local source is being used.

Reviewer: ```suggestion
> [!IMPORTANT]
> On windows it might be necessary to manually create the VERSION file
> when a local source is being used.
```

Using the syntax from https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

To enable this feature:

```bash
cargo build --features=bundled
```

```bash
TPM2_TSS_VERSION="4.1.3" cargo build --features=bundled
```

```bash
TPM_TSS_SOURCE_PATH=/path/to/tpm-tss cargo build --features=bundled
```

If using this feature from an external project

```

> tss-esapi-sys = { version = "...", features = "bundled" }

Reviewer: ```suggestion
tss-esapi-sys = { version = "...", features = ["bundled"] }
```

Not sure if a bare string is accepted but I think it need to be an array :thinking:

```

### Windows

Compiling for windows requires a bit of setup to work with the bundled feature.

> * Openssl must be installed to a non-standard location at C:\OpenSSL-v11-Win64

Reviewer: ```suggestion
* OpenSSL must be installed to a non-standard location at `C:\OpenSSL-v11-Win64`
```

> * Visual studio 2019 must be installed with the Clang/C2 experimental component,

Reviewer: ```suggestion
* Visual Studio 2019 must be installed with the Clang/C2 experimental component,
```
